scripts/hid_configurator/configurator_core.py

```diff
@@ -620,5 +623,74 @@ def is_dfu_image_correct(dfu_image):
     return True
 
 
+class Step:
+    def __init__(self, r, g, b, substep_count, substep_time):
+        self.r = int(r)
+        self.g = int(g)
+        self.b = int(b)
+        self.substep_count = int(substep_count)
+        self.substep_time = int(substep_time)
+
+
+def led_send_single_step(dev, recipient, step):
+    event_id = (EVENT_GROUP_LED_STREAM << GROUP_FIELD_POS) | (STREAM_DATA << TYPE_FIELD_POS)
+
+    event_data = struct.pack('<BBBHH', step.r, step.g, step.b, step.substep_count, step.substep_time)
+
+    try:
+        success = exchange_feature_report(dev, recipient, event_id, event_data, False)
+    except:
+        success = False
+
+    return success
+
+
+def led_send_random_color_steps(dev, recipient, n):
+    logging.debug('Sending random color steps')
+    substep_count = 10
```

Comment:
Maybe we could define these default values e.g. in `__init__`?

Reply:
step substep_count and substep_time will change dynamically, so I don't want them in __init__